directjngine/src/main/java/com/softwarementors/extjs/djn/servlet/DirectJNgineServlet.java

```diff
@@ -657,4 +676,31 @@ private void processUploadFormPost(HttpServletRequest request, HttpServletRespon
     return uploader.parseRequest(request);
   }
 
+  private boolean isValidAntiCsrfTokenMatching(final HttpServletRequest request, final List<FileItem> fileItems) {
+    String cookie = getAntiCsrfTokenCookie(request);
+    String form = getAntiCsrfTokenField(fileItems);
+
+    return !antiCsrfTokenEnabled || (cookie != null && cookie.equals(form));
+  }
+
+  private String getAntiCsrfTokenCookie(final HttpServletRequest request) {
+    Cookie[] cookies = request.getCookies();
+    if (cookies != null) {
+      for (Cookie cookie : cookies) {
+        if (ANTI_CSRF_TOKEN_NAME.equals(cookie.getName())) {
+          return cookie.getValue();
+        }
+      }
+    }
+    return null;
+  }
+
+  private String getAntiCsrfTokenField(final List<FileItem> fileItems) {
+    for (FileItem item : fileItems) {
+      if (item.isFormField() && ANTI_CSRF_TOKEN_NAME.equals(item.getFieldName())) {
```

Comment:
All fields become `FileItem`s? Seems odd.

Reply:
That's correct. They are later separated into files and fields [here](https://github.com/sonatype/directjngine/blob/fb7e513b4d02eaa0cbc15e237cf1f935188cf895/directjngine/src/main/java/com/softwarementors/extjs/djn/router/processor/standard/form/upload/UploadFormPostRequestProcessor.java#L62)